Lib/ufo2ft/instructionCompiler.py

```diff
@@ -0,0 +1,194 @@
+
+                # Set component flags
+
+                # We need to decide when to set the flags.
+                # Let's assume if any lib key is not there, or the component
+                # doesn't have an identifier, we should leave the flags alone.
+                use_my_metrics_comp = None
+                for i, c in enumerate(glyf.components):
+                    if i >= len(glyph.components):
+                        logger.error(
+                            "Number of components differ between UFO and TTF "
+                            f"in glyph '{name}' ({len(glyph.components)} vs. "
+                            f"{len(glyf.components)}, not setting flags in "
+                            "additional components."
+                        )
+                        break
+                    ufo_component_id = glyph.components[i].identifier
+                    if (
+                        ufo_component_id is not None
+                        and OBJECT_LIBS_KEY in glyph.lib
+                        and ufo_component_id in glyph.lib[OBJECT_LIBS_KEY]
+                        and (
+                            TRUETYPE_ROUND_KEY
+                            in glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+                            or TRUETYPE_METRICS_KEY
+                            in glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+                        )
+                    ):
+                        component_lib = glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+
+                        c.flags &= ~ROUND_XY_TO_GRID
+                        if component_lib.get(TRUETYPE_ROUND_KEY, False):
+                            c.flags |= ROUND_XY_TO_GRID
+
+                        c.flags &= ~USE_MY_METRICS
+                        if component_lib.get(TRUETYPE_METRICS_KEY, False):
+                            if use_my_metrics_comp:
+                                logger.warning(
+                                    "Ignoring USE_MY_METRICS flag on component "
+                                    f"'{ufo_component_id}' because it has been set on "
+                                    f"component '{use_my_metrics_comp}' already."
+                                )
+                            else:
+                                c.flags |= USE_MY_METRICS
+                                use_my_metrics_comp = ufo_component_id
+
+                    # We might automatically set the flags if no data is present,
+                    # but:
+                    # - https://github.com/googlefonts/ufo2ft/pull/425 recommends
+                    #   against setting the ROUND_XY_TO_GRID flag
+                    # - USE_MY_METRICS has been set already by
+                    #   outlineCompiler.OutlineTTFCompiler.autoUseMyMetrics
+
+                    if i == 0 and TRUETYPE_OVERLAP_KEY in glyph.lib:
+                        # Set OVERLAP_COMPOUND on the first component only
+                        c.flags &= ~OVERLAP_COMPOUND
```

Comment:
why not putting this line in an `else:` block below; feelds slightly weird to me that you first unconditionally un-set it and then set it if True..